asserts/fetcher.go

```diff
@@ -106,6 +135,44 @@ func (f *fetcher) Fetch(ref *Ref) error {
 	return f.chase(ref, nil)
 }
 
+func (f *fetcher) isSeqFetched(seq *AtSequence) (bool, error) {
+	u := seq.Unique()
+	switch f.fetched[u] {
+	case fetchSaved:
+		return true, nil // nothing to do
+	case fetchRetrieved:
+		return false, fmt.Errorf("circular assertions are not expected: %s", seq)
+	}
+	return false, nil
+}
+
+func (f *fetcher) fetchSequence(seq *AtSequence) error {
+	// sequence forming assertions are never predefined, so we don't check for it.
+	if ok, err := f.isSeqFetched(seq); err != nil || ok {
+		// if ok is true, then the assertion was fetched and err is nil
+		return err
+	}
+	retrieved, err := f.retrieveSeq(seq)
+	if err != nil {
+		return err
+	}
+	u := seq.Unique()
+	f.fetched[u] = fetchRetrieved
```

Comment:
sequence forming assertions could also have prerequisites so the code starting here should be shared with chase